src/dbnode/storage/index/block.go

```diff
@@ -975,6 +1000,48 @@ func (b *block) EvictMutableSegments() error {
 	return multiErr.FinalError()
 }
 
+func (b *block) NeedsColdMutableSegmentsEvicted() bool {
+	b.RLock()
+	defer b.RUnlock()
+	var anyColdMutableSegmentNeedsEviction bool
+	for _, coldSeg := range b.coldMutableSegments {
+		anyColdMutableSegmentNeedsEviction = anyColdMutableSegmentNeedsEviction || coldSeg.NeedsEviction()
+	}
+	return anyColdMutableSegmentNeedsEviction
+}
+
+func (b *block) EvictColdMutableSegments() error {
+	b.Lock()
+	defer b.Unlock()
+	if b.state != blockStateSealed {
+		return fmt.Errorf("unable to evict cold mutable segments, block must be sealed, found: %v", b.state)
+	}
+
+	// Evict/remove all but the most recent cold mutable segment (That is the one we are actively writing to).
+	for i, coldSeg := range b.coldMutableSegments {
+		if i < len(b.coldMutableSegments)-1 {
+			coldSeg.Evict()
+			b.coldMutableSegments[i] = nil
+		}
+	}
+	// Swap last w/ first and truncate the slice.
+	lastIdx := len(b.coldMutableSegments) - 1
+	b.coldMutableSegments[0], b.coldMutableSegments[lastIdx] = b.coldMutableSegments[lastIdx], b.coldMutableSegments[0]
```

Comment:
Probably no need to swap the head with the tail, just need the tail to become the head yeah?
```
b.coldMutableSegments[0] = b.coldMutableSegments[len(b.coldMutableSegments) - 1]
b.coldMutableSegments = b.coldMutableSegments[:1]
```